### PR TITLE
remove joda time usage from presto-redis module

### PR DESCRIPTION
**Pull request by dutta-sh**
Test plan - mvn clean test from command line:
[INFO] Tests run: 857, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 1,307.459 s - in TestSuite
[INFO] 
[INFO] Results:
[INFO] 
[INFO] Tests run: 857, Failures: 0, Errors: 0, Skipped: 0
[INFO] 
[INFO] ------------------------------------------------------------------------
[INFO] BUILD SUCCESS
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  22:19 min
[INFO] Finished at: 2022-07-12T17:23:28-04:00
[INFO] ------------------------------------------------------------------------ 

Please make sure your submission complies with our [Development](https://github.com/prestodb/presto/wiki/Presto-Development-Guidelines#development), [Formatting](https://github.com/prestodb/presto/wiki/Presto-Development-Guidelines#formatting), and [Commit Message](https://github.com/prestodb/presto/wiki/Review-and-Commit-guidelines#commit-formatting-and-pull-requests) guidelines. Don't forget to follow our [attribution guidelines](https://github.com/prestodb/presto/wiki/Review-and-Commit-guidelines#attribution) for any code copied from other projects.

Fill in the release notes towards the bottom of the PR description.
See [Release Notes Guidelines](https://github.com/prestodb/presto/wiki/Release-Notes-Guidelines) for details.

```
== NO RELEASE NOTE ==
```

### CODE REVIEW

File: presto-redis/src/main/java/com/facebook/presto/redis/decoder/hash/ISO8601HashRedisFieldDecoder.java [hunk 1]
```diff
@@ -51,7 +52,7 @@ public ISO8601HashRedisValueProvider(DecoderColumnHandle columnHandle, String va
         @Override
         public long getLong()
         {
-            long millis = FORMATTER.parseMillis(getSlice().toStringAscii());
+            long millis = LocalDate.from(FORMATTER.parse(getSlice().toStringAscii())).atStartOfDay().toInstant(ZoneOffset.UTC).toEpochMilli();
```
Comment (ajaygeorge) [inline presto-redis/src/main/java/com/facebook/presto/redis/decoder/hash/ISO8601HashRedisFieldDecoder.java:55]: `atStartOfDay` feels like loss of fidelity. Can you try passing in a timestamp which has hr, min in it and see if the test succeeds.

Comment (dutta-sh) [reply presto-redis/src/main/java/com/facebook/presto/redis/decoder/hash/ISO8601HashRedisFieldDecoder.java:55]: `LocalDate` would not parse a field that has Date and time. More here - https://stackoverflow.com/questions/27454025/unable-to-obtain-localdatetime-from-temporalaccessor-when-parsing-localdatetime

We can create a custom formatter with `parseDefaulting` for the time part I guess. Please suggest

Comment (ajaygeorge) [reply presto-redis/src/main/java/com/facebook/presto/redis/decoder/hash/ISO8601HashRedisFieldDecoder.java:55]: Considering using a class like `LocalDateTime` which has time component.
https://docs.oracle.com/javase/8/docs/api/java/time/LocalDateTime.html

Comment (dutta-sh) [reply presto-redis/src/main/java/com/facebook/presto/redis/decoder/hash/ISO8601HashRedisFieldDecoder.java:55]: Had already tried `LocalDateTime` with `DateTimeFormatter.ISO_LOCAL_DATE_TIME` as well with `DateTimeFormatter.ISO_LOCAL_DATE`

- With `ISO_LOCAL_DATE_TIME` the error is - `Text '1992-12-22' could not be parsed at index 10`
- With `ISO_LOCAL_DATE` the error is - `Unable to obtain LocalDateTime from TemporalAccessor: {},ISO,UTC resolved to 1992-11-11 of type java.time.format.Parsed`

Comment (dutta-sh) [reply presto-redis/src/main/java/com/facebook/presto/redis/decoder/hash/ISO8601HashRedisFieldDecoder.java:55]: Using `formatter.parseBest` would fix the issue. Will update

**Comment by dutta-sh** [comment]
- taken care of in https://github.com/prestodb/presto/pull/18035
- closing this one due to EasyCLA issues with git id

**Pull request closed without merging**